### DISCUSSION TITLE
Multitenant solution on same computer

### DISCUSSION THREAD

**Discussion by LuisVilaplana** [oqtane/oqtane.framework#3265] [General] [answered]
Hello, I have been having problems with a multi tenant installation. The solution is for a client that has several branches, one for each site. They require each site to be open on different windows or tabs on same computer. I have tried several ways, including installing each site as a different application in MS Edge.

The problem I have is the following:

User1 opens site1, but also needs to open site2 in another window. When you have both windows open, when you try to change pages everything starts to crash. Even if he tries to log out, both sites present a blank page and nothing more can be done. Even if I restart the application in Visual Studio, the problem continues and the only way to reset everything is to do a Clean Solution.

Tried this on IIS on a web server too with same results.

I hope this explanation is clear.

Thank you so much

**Comment by sbwalker** [comment]
@LuisVilaplana having a single installation of Oqtane on a server and allowing multiple users to access the various sites which are part of the installation is the core capability which Oqtane provides. There are many production installations of Oqtane actively using this capability. It sounds like a configuration issue in your local environment. You will need to provide more information on how you defined your tenant Urls, etc...

**Reply by LuisVilaplana** [reply]
In development, using Visual Studio 2022, SQL Server Developer, and IIS Express, I set up sites this way:

Main site: localhost:44357
Site 1: localhost:44357/site1
Site 2: localhost:44357/site2

On production Windows Server 2019, SQL Server Express and IIS, I set up sites this way:

Main site: https://www.prosoftpos.net
Site 1: https://www.prosoftpos.net/client1
Site 2: https://www.prosoftpos.net/client2

I didn't try with subdomains. I'm using a separate database per site

Regards

**Reply by sbwalker** [reply]
Based on the url configuration you shared, both sites are using the same domain. This means the auth cookies are shared between the different browser tabs you have open (because they belong to the same domain). As you perform activities in one tab it may affect the cookies being used in the other tab. The router contains logic to detect if the site alias has changed, and if so it reloads the auth principal. I believe this is why you are experiencing strange behavior as you switch between tabs. If you were using distinct domains or subdomains for each site in your installation the cookies would not be shared and you would not experience this behavior.

**Reply by LuisVilaplana** [reply]
Ok, on production server I configured both sites like this:

![image](https://github.com/oqtane/oqtane.framework/assets/103559699/97cc3a44-50ac-4c46-ad36-a11d9068d9ed)

But a cannot make it work in Plesk. I asked this on a separate discussion.

On the other hand, how can I simulate this in my development environment using localhost?

Thanks

**Comment by LuisVilaplana** [comment, marked as answer]
I solved this with your help on production server! Please refer to discusion #3281

Once I was able to resolve subdomain name resolution in Plesk and IIS this issue resolved itself. Oqtane does a great job of separating Tenants without having to configure anything else.

I only have to solve it on localhost in my development environment, but I already have an idea based on another discussion.

Thank you very much for your help! I am happy to be able to integrate Oqtane into my solutions. I love Blazor and I'm an old DNN user from the old days!

Regards
